BigBaseV2/src/views/network/view_session.cpp

```diff
@@ -1,17 +1,64 @@
 #include "fiber_pool.hpp"
 #include "util/session.hpp"
 #include "views/view.hpp"
+#include "pointers.hpp"
+#include "gta/joaat.hpp"
+#include "util/notify.hpp"
+#include "rage/rlSessionByGamerTaskResult.hpp"
 
 namespace big
 {
+	static void join_by_rockstar_id(uint64_t rid)
+	{
+		if (SCRIPT::GET_NUMBER_OF_THREADS_RUNNING_THE_SCRIPT_WITH_THIS_HASH(RAGE_JOAAT("maintransition")) != 0 ||
+			STREAMING::IS_PLAYER_SWITCH_IN_PROGRESS())
+		{
+			notify::above_map("Cannot RID join now");
```

Comment:
Might get obstructed by the player list rendering over this, prefer using the notification service.